src/v/wasm/allocator.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/aligned_buffer.hh>
+#include <seastar/core/chunked_fifo.hh>
+#include <seastar/util/optimized_optional.hh>
+
+#include <type_traits>
+
+namespace wasm {
+
+/**
+ * An owned bit of heap memory, which is aligned to page size and it's size is a
+ * multiple of the page size.
+ */
+struct heap_memory {
+    std::invoke_result_t<
+      decltype(ss::allocate_aligned_buffer<uint8_t>),
+      size_t,
+      size_t>
```

Comment:
is this temporary_buffer<uint8_t> ?

Reply:
No `std::unique_ptr<uint8_t[], ss::free_deleter>` I've explicitly added the type.